### ISSUE TITLE
Keep maximum network message in the NetAddr

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#5241]
## Motivation

Currently there are several places in the code where the maximum (UDP) network packet size is tested for, which is hardcoded.

See discussions here:
- https://github.com/supercollider/supercollider/pull/5155
- https://github.com/supercollider/supercollider/pull/5238

## Description of Proposed Feature

Ask the `NetAddr` for the maximum packet size. This can be configurable as well as it can depend on the protocol used (e.g. for TCP).

## Plan for Implementation

We need to implement:
- a way to know whether a `NetAddr` is connected via TCP
- a settable instance variable in `NetAddr` called `maxMessageByteSize`
- review all the places where we currently have some hardcoded number

Open questions:
- what should the number actually be (see https://github.com/supercollider/supercollider/pull/5155) ? 
- Maybe we need a way to determine the size of some part of the message, like `"/n_setn", 651` so that it can be subtracted from that maximum number.

Once these questions are clear, I can implement it.

**Comment by brianlheim** [comment]
> a way to know whether a NetAddr is connected via TCP

does isConnected not serve this purpose already?

> a settable instance variable in NetAddr called maxMessageByteSize

i don't know much about networking, is this for UDP or TCP or both?

**Comment by brianlheim** [comment]
> Maybe we need a way to determine the size of some part of the message, like "/n_setn", 651 so that it can be subtracted from that maximum number.

if we don't have a primitive to compute OSC message length, without actually constructing a message, i think it would be a good addition.

**Comment by telephon** [comment]
> if we don't have a primitive to compute OSC message length, without actually constructing a message, i think it would be a good addition.

definitely.

> > a way to know whether a NetAddr is connected via TCP
> 
> does isConnected not serve this purpose already?
> 

I am not sure, but it looks that way indeed.

> > a settable instance variable in NetAddr called maxMessageByteSize
> 
> i don't know much about networking, is this for UDP or TCP or both?

As far as I know there is no definite limit on TCP, because it is streamed chunk by chunk. But I am no expert at all, so I am unsure.

**Comment by smrg-lm** [comment]
I made the calculation in python, you can take it if you want, also i use udp max size instead an arbitrary number.

**Comment by smrg-lm** [comment]
And just for the record, you can take anything you want if you see it fit, is the same library.

**Comment by telephon** [comment]
Yes, let's try it out. It is a good first step to have one single place for it.

Probably it would be easy for some of us to write a primitive for the message size (as @brianlheim mentions above).
